### ISSUE TITLE
Using -t with special characters

### ISSUE THREAD

**Issue by erikng** [sheagcraig/yo#1]
/Applications/Utilities/yo.app/Contents/MacOS/yo -t "Test!"

fails with -bash: !": event not found

**Comment by sheagcraig** [comment]
@erikng Indeed, that's by design. Kind of.

That's really more a bash-ism than anything else. The `!` character in bash is the history operator. It allows you to replace the ! character with some previous operation.

Type `history` and you get the bash history, with numbers indicating the history number. So you can replay a command with `!<and that number>` e.g. `!365`. And much much more.

So you have to wrap your title in single quotes rather than double-quotes. Double-quoted stuff still allows variable expansion, history expansion, etc. So `"Test!"` is really like saying Test and then a history operation.

Single quoted stuff is strictly what's quoted. So try `'Test'`.

Indeed, I haven't actually tried it, but the CommandLine.framework which I use for the argument processing accepts Unicode, so you should be able to put emojis and stuff in.

**Comment by erikng** [comment]
Single quoted works like a charm, but I couldn't get emoji to work.

**Comment by sheagcraig** [comment]
You can drag an emoji from the Special Characters palette. That's the easiest way.

Otherwise, if you need the hex code, find it in the Special Characters and go to bash:

```
echo -ne '<drag the emoji here>' | hexdump
```

This will print out the hex code for the emoji. It's the rightmost column of values that are in hex. You have to slap a \x before each two-characters...

and then you can echo it back up as a command substitution...

Complete process:

```
$ echo -ne '💩' | hexdump
0000000 f0 9f 92 a9                                    
0000004

# So 'f09f92a9' is the hex value...
$ /Applications/Utilities/yo.app/Contents/MacOS/yo -t $(echo -ne '\xf0\x9f\x92\xa9')
```

= Smiling poo emoji notification.

You can also do `printf '\xf0\x9f\x92\xa9'` as well.

**Comment by erikng** [comment]
I was simply dragging from the palette and figured that wouldn't work. 

Thanks for explaining. The emoji is a cool idea that I hadn't thought about. I probably won't use it for my initial use case, but perhaps if I think of some other method. I did tag you on Twitter for a post I did yesterday with my initial use case. 

I'd like to somehow trigger these alerts through a sever ("push notification" style) but I haven't given it enough thought yet.
